docs/tests/core-concepts/container-images/simple/build.cue

```diff
@@ -29,3 +36,12 @@ dagger.#Plan & {
 		]
 	}
 }
+
+// Example usage in a plan
+dagger.#Plan & {
+	client: filesystem: "./src": read: contents: dagger.#FS
+
+	actions: build: #PythonBuild & {
+		app: client.filesystem."./src".read.contents
+	}
+}
```

Comment:
Unidirectional flow comes easy to me: top to bottom, left to write.

By extracting the `#PythonBuild` definition, a jump from line 45 (bottom) to line 10 (top) is forced, which breaks the unidirectional flow. I find this less easy to read & understand.

Same applies to `with-dockerfile.cue` for me.

Reply:
Agreed

Reply:
Thanks for your input.

I disagree. I’m seeing people confused by examples that focus too much on the plan and not enough on the ability to create your own actions. Actions hardcoded in the plan don’t show the power of dagger: it kind of looks like an over engineered makefile. We have to emphasize a workflow where first you create your actions, then you compose them.